src/_pytest/recwarn.py

```diff
@@ -287,27 +294,39 @@ def __exit__(
 
         __tracebackhide__ = True
 
+        if self.expected_warning is None:
+            # nothing to do in this deprecated case, see WARNS_NONE_ARG above
+            return
+
+        if not (exc_type is None and exc_val is None and exc_tb is None):
+            # We currently ignore missing warnings if an exception is active.
+            # TODO: fix this, because it means things are surprisingly order-sensitive.
```

Comment:
Not sure what this TODO means here... can we clarify it? Also not clear if this is something meant to be done still in this PR, or later.

Reply:
Ah, that's a note-to-self about rebasing https://github.com/pytest-dev/pytest/pull/11129 which I'll do immediately after this is merged.